plugins/flatpak/gs-flatpak.c

```diff
@@ -2891,13 +2891,14 @@ gs_flatpak_add_app_to_list_if_not_installed (GsApp *app, GsAppList *list,
 					     const GHashTable *hash_installed)
 {
 	g_autofree gchar *ref_display = gs_flatpak_app_get_ref_display (app);
-	if (g_hash_table_contains (hash_installed, ref_display)) {
-		g_debug ("%s is already installed, so skipping",
+	if (g_hash_table_contains (hash_installed, ref_display) &&
+	    !gs_app_is_updatable (app)) {
```

Comment:
I am not sure about this.
Does this mean that any installation/update may try to update the runtimes and services app too? Before we used to keep the update only for the proxy app, so maybe it's better to keep it that way until we revise this when we start adding the "size" as a criteria for Mogwai.

In any case, such a change should also change how the function is called as it's now misleading.

This code if kind of critical/delicate as it may stop installations and updates from working, so I prefer to be conservative here. Thus, if possible let's remove this part and open a task for the future if runtimes and services should be updated together with the apps.

Reply:
At the moment, flatpak on the CLI always upgrades any related refs and the runtimes whenever you install or upgrade an app. Does GNOME software not do this? It could cause subtle bugs (for instance LibreOffice 6 needs some settings schemas for HiDPI in Wayland which were only added more recently to FD.o).

Anyway, this function is Endless specific, it's only called in the code that considers adding the SDK to the list of apps to install.

In any case, if we don't upgrade EKSM, this PR isn't worthwhile - the OS makes it nearly impossible for us to not have EKSM installed. The situation we want to avoid is you install your first SDK4 app but your EKSM is old and doesn't know how to access those files. That's the primary goal of keeping this code in GS - otherwise we could drop it all.

Reply:
g-s does update related apps together with other apps. But it doesn't update runtimes when updating an app (if there's a new runtime/runtime-branch needed for the new app update, then it's installed though). Just to be clear, I am fine if we need to update the services app (which may be relatively small) together with knowledge apps, but your patch affects also runtimes, which I am not sure should be treated that way here.
We have some logic in the auto updater to consider app updates first, if now every app ends up updating the runtimes too, that logic doesn't make much sense (besides requiring more testing, etc.).
All this makes me feel safer if we leave the runtimes outside of this ATM (after the release we can discuss if this is a good thing with upstream as well).

Reply:
Oh interesting. The dangers of mobile github. I saw in Philip's old commit that this function was new but didn't catch it was still used in the runtime case (which it was factored out of). So yeah, I hadn't intended to change the runtime behaviour indeed, that's a separate debate. Arguably not necessary for flatpak runtimes as a branch is meant to be an ABI... but... well. We should fix this so it only updates EKSM in the knowledge app case. For a quick fix we can just drop this code and it will be OK until SDK5 appears. :)